packages/ld_default.rb

```diff
@@ -2,54 +2,70 @@
 
 class Ld_default < Package
   description 'Select the default ld executable and check for libraries in #{CREW_LIB_PREFIX} first'
-  homepage 'https://github.com/skycocker/chromebrew'
-  version '1.0'
-  source_url 'https://raw.githubusercontent.com/skycocker/chromebrew/93aa4dc4d9198e28d84dcf9d59b3ab6b6a94d92c/README.md'
-  source_sha256 '3ec9c6f7ba6a138d07dfb4a372d562cd3ad558b58b6bd447c2dbb52e16b697a5'
+  homepage 'https://github.com/skycocker/chromebrew/wiki/FAQ'
+  version '1.1'
+  source_url 'file:///dev/null'
+  source_sha256 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855'
 
   depends_on 'llvm'
 
   def self.build
-system "cat << 'EOF' > ld_default
+    system "cat << 'EOF' > ld_default
 #!/bin/bash
-cd #{CREW_PREFIX}/bin
-type=\$(file ld | cut -d' ' -f2 | tr -d '\\n')
-if [ \"\$type\" == \"ELF\" ]; then
-  current=\$(find . -inum $(ls -i ld 2> /dev/null | cut -d' ' -f1) 2> /dev/null | fgrep 'ld.' 2> /dev/null | cut -d/ -f2 2> /dev/null)
-else
-  slashes=\$(tr -dc '/' <<<'#{CREW_PREFIX}/bin/ld/' | wc -c)
-  current=\$(tail -1 ld | cut -d'/' -f\${slashes} 2> /dev/null | cut -d' ' -f1 2> /dev/null)
-fi
-echo
-echo \"Current default: \$current\"
-echo
-echo 'Enter the ld default:'
-echo 'b = ld.bfd'
-echo 'g = ld.gold'
-echo 'l = ld.lld'
-echo '0 = Cancel'
-while true; do
-  read default
-  case $default in
-    b)
-      current='ld.bfd'
-      break;;
-    g)
-      current='ld.gold'
-      break;;
-    l)
-      current='ld.lld'
-      break;;
-    0)
-      exit;;
-    *)
-      echo 'Please select from one of the options or enter 0 to cancel.'
-  esac
-done
-echo '#!/bin/bash' > /tmp/ld
-echo '#{CREW_PREFIX}/bin/'$current' --library-path #{CREW_LIB_PREFIX} -rpath #{CREW_LIB_PREFIX} \"\$@\"' >> /tmp/ld
-install -Dm755 /tmp/ld #{CREW_PREFIX}/bin/ld
-rm -f /tmp/ld
+
+  set -e
+
+  mktmp=\$(mktemp -d)
+
+  cd #{CREW_PREFIX}/bin
+
+  type=\$(file ld | cut -d' ' -f2 | tr -d '\\n')
+
+  if [ \"\${type}\" == \"ELF\" ]; then
+    current=\$(find . -inum $(ls -i ld 2>/dev/null | cut -d' ' -f1) 2>/dev/null | fgrep 'ld.' 2>/dev/null)
+  elif [ \"${type}\" == \"symbolic\" ]; then
+    current=\$(basename `readlink ld`)
+  else
+    current=\$(tail -1 ld | cut -d' ' -f1 2>/dev/null)
```

Comment:
There are 2 things that can make this better:
```
if [ \"\${type}\" == \"ELF\" ]; then
     current=\$(find . -inum $(ls -i ld 2>/dev/null | cut -d' ' -f1) 2>/dev/null | fgrep 'ld.' 2>/dev/null)
   elif [ \"${type}\" == \"symbolic\" ]; then
     current=\$(readlink ld)
   else
     current=\$(basename \$(tail -1 ld | cut -d' ' -f1))
   fi
```
The readlink command already strips the full path.  The symlink scenario will only happen if it is created manually.  If somebody already knows how to do that, they don't need this script.

Reply:
```
chronos@localhost ~/Downloads $ ln -s /usr/local/bin/hello file
chronos@localhost ~/Downloads $ ls -la file
lrwxrwxrwx. 1 chronos chronos 22 Dec  3 14:58 file -> /usr/local/bin/hello
chronos@localhost ~/Downloads $ readlink file
/usr/local/bin/hello
chronos@localhost ~/Downloads $
```
True.

Reply:
What do you want me to change?

Reply:
Your readlink works differently than mine.  OK, I guess leave basename but please use $(...) instead of `...`.  Backticks is bad form.

Reply:
Oh. So you do want to keep the symlink functionality or you don't?

Reply:
Yes, we can keep it but it will never happen to anybody except yourself. :)

Reply:
I re-added it with `$()` instead of backticks. Is it good now?

Reply:
Still need `basename` on the last conditional.

Reply:
You can also remove all the `2>/dev/null`s.  I cannot think of a situation where it would error.

Reply:
I can. If 'ld' is not there or if it's a broken link. If that errors, then `find` will fail because it's not trying to find anything. If that fails, `fgrep` will go forever, as it's trying to read broken input.

Reply:
This looks good right now. After this is merged, `basename` will never be needed on the last conditional. Also, don't backslash out the '$'.

Reply:
Please use basename on the last line.  That is a full path to ld.
```
current=\$(basename \$(tail -1 ld | cut -d' ' -f1))
```

Reply:
If find fails, we want to know why, correct?  Please remove.

Reply:
All right. Done.